pkg/controller/v1alpha2/core/scopes/healthscope/healthscope.go

```diff
@@ -394,3 +369,65 @@ func getVersioningPeerWorkloadRefs(ctx context.Context, c client.Reader, wlRef r
 	}
 	return peerRefs, nil
 }
+
+// PeerHealthConditions refers to a slice of health condition of worloads
+// belonging to one version-enabled component
+type PeerHealthConditions []WorkloadHealthCondition
+
+func (p PeerHealthConditions) Len() int { return len(p) }
+func (p PeerHealthConditions) Less(i, j int) bool {
+	// sort by revision number in descending order
+	return extractRevision(p[i].TargetWorkload.Name) > extractRevision(p[j].TargetWorkload.Name)
+}
+func (p PeerHealthConditions) Swap(i, j int) { p[i], p[j] = p[j], p[i] }
+
+// exract revision number from revision name in format: <comp-name>-v<revision number>
+// any non-qualified format should return 0
+func extractRevision(c string) int {
+	i, _ := strconv.ParseInt(c[strings.LastIndex(c, "v")+1:], 10, 0)
+	return int(i)
+}
+
+// MergePeerWorkloadsConditions merge health conditions of all peer workloads into basic
+func (p PeerHealthConditions) MergePeerWorkloadsConditions(basic *WorkloadHealthCondition) {
+	if basic == nil || len(p) == 0 {
+		return
+	}
+	// copy to keep idempotent
+	peerHCs := make(PeerHealthConditions, len(p))
+	copy(peerHCs, p)
+	peerHCs = append(peerHCs, *basic.DeepCopy())
+
+	// sort by revision number in descending order
+	sort.Sort(peerHCs)
+
+	for _, peerHC := range peerHCs {
+		if peerHC.HealthStatus == StatusUnhealthy {
+			// if ANY peer workload is unhealthy
+			// then the overall condition is unhealthy
+			basic.HealthStatus = StatusUnhealthy
+		}
+	}
+	// re-format diagnosis/workloadStatus to show multiple workloads'
+	if basic.HealthStatus == StatusUnknown {
+		basic.WorkloadStatus = fmt.Sprintf("%s:%s", peerHCs[0].TargetWorkload.Name, peerHCs[0].WorkloadStatus)
+		for _, peerHC := range peerHCs[1:] {
+			basic.WorkloadStatus = fmt.Sprintf("%s %s:%s",
+				basic.WorkloadStatus,
+				peerHC.TargetWorkload.Name,
+				peerHC.WorkloadStatus)
+		}
+	} else {
+		basic.Diagnosis = fmt.Sprintf("%s:%s", peerHCs[0].TargetWorkload.Name, peerHCs[0].Diagnosis)
+		for i, peerHC := range peerHCs[1:] {
+			if i > 1 && peerHC.Diagnosis == fmt.Sprintf(infoFmtReady, 0, 0) {
```

Comment:
```suggestion
			if i > 0 && peerHC.Diagnosis == fmt.Sprintf(infoFmtReady, 0, 0) {
```

I hope we just keep one

Reply:
Sure.